test/model_simulation/simulate_jumps.jl

```diff
@@ -91,56 +103,60 @@ let
     jump_3_5 = ConstantRateJump(rate_3_5, affect_3_5!)
     jump_3_6 = ConstantRateJump(rate_3_6, affect_3_6!)
     jumps_3 = (jump_3_1, jump_3_2, jump_3_3, jump_3_4, jump_3_5, jump_3_6)
-    push!(identical_networks, reaction_networks_constraint[5] => jumps_3)
+    push!(catalyst_networks, reaction_networks_constraint[5])
+    push!(manual_networks, jumps_3)
+    push!(u0_syms, [:X1, :X2, :X3, :X4])
+    push!(ps_syms, [:k1, :k2, :k3, :k4, :k5, :k6])
+
+    # Loops through all cases, checks that identical simulations are generated with/without Catalyst.
+    for (rn_catalyst, rn_manual, u0_sym, ps_sym) in zip(catalyst_networks, manual_networks, u0_syms, ps_syms)
+        for factor in [5, 50]
+            u0_1 = rnd_u0_Int64(rn_catalyst, rng; n = factor)
+            ps_1 = rnd_ps(rn_catalyst, rng; factor = factor/100.0)
+            dprob_1 = DiscreteProblem(rn_catalyst, u0_1, (0.0, 100.0), ps_1)
+            jprob_1 = JumpProblem(rn_catalyst, dprob_1, Direct(); rng)
+            sol1 = solve(jprob_1, SSAStepper(); seed = 1234, saveat = 1.0)
+            
+            u0_2 = map_to_vec(u0_1, u0_sym)
+            ps_2 = map_to_vec(ps_1, ps_sym)
+            dprob_2 = DiscreteProblem(u0_2, (0.0, 100.0), ps_2)
+            jprob_2 = JumpProblem(dprob_2, Direct(), rn_manual...; rng)
+            sol2 = solve(jprob_2, SSAStepper(); seed = 1234, saveat = 1.0)
 
-    for (i, networks) in enumerate(identical_networks)
-        for factor in [1e-2, 1e-1, 1e0, 1e1]
-            (i == 3) && (factor > 1e-1) && continue   # Large numbers seems to crash it.
-            u0 = rand(rng, 1:Int64(factor * 100), length(get_unknowns(networks[1])))
-            p = factor * rand(rng, length(get_ps(networks[1])))
-            prob1 = JumpProblem(networks[1],
-                                DiscreteProblem(networks[1], u0, (0.0, 1000.0), p),
-                                Direct())
-            sol1 = solve(prob1, SSAStepper())
-            prob2 = JumpProblem(DiscreteProblem(u0, (0.0, 1000.0), p), Direct(),
-                                networks[2]...)
-            sol2 = solve(prob2, SSAStepper())
-            for i in 1:length(u0)
-                vals1 = getindex.(sol1.u, i)
-                vals2 = getindex.(sol1.u, i)
-                (mean(vals2) > 0.001) && @test 0.8 < mean(vals1) / mean(vals2) < 1.25
-                (std(vals2) > 0.001) && @test 0.8 < std(vals1) / std(vals2) < 1.25
+            if nameof(rn_catalyst) == :rnh7
```

Comment:
Still no idea what happens with this one case. I have probably spent 3 hours trying to figure it out. Really got no idea at this stage.

Reply:
Can you give me a standalone MWE on this example that breaks to play with? I'm not even sure what network this is. (i.e. that I can just cut and paste and doesn't require pulling a network out of another file.)

Reply:
Yes, I will do this. I will just try to do it as minimal as possible before I send it to you (which in this case turned out surprisingly hard)